MdeModulePkg/Bus/Pci/NvmExpressDxe/NvmExpress.c

```diff
@@ -166,6 +167,26 @@ EnumerateNvmeDevNamespace (
     Device->BlockIo2.FlushBlocksEx = NvmeBlockIoFlushBlocksEx;
     InitializeListHead (&Device->AsyncQueue);
 
+    //
+    // Create Media Sanitize Protocol instance
+    //
+    Device->MediaSanitize.Revision    = MEDIA_SANITIZE_PROTOCOL_REVISION;
+    Device->MediaSanitize.Media       = &Device->Media;
+    Device->MediaSanitize.MediaClear  = NvmExpressMediaClear;
+    Device->MediaSanitize.MediaPurge  = NvmExpressMediaPurge;
+    Device->MediaSanitize.MediaFormat = NvmExpressMediaFormat;
+
+    ASSERT (
+      sizeof (Device->MediaSanitize.SanitizeCapabilities) ==
+      sizeof (Device->Controller->ControllerData->Sanicap)
+      );
+
+    CopyMem (
+      &(Device->MediaSanitize.SanitizeCapabilities),
+      &(Device->Controller->ControllerData->Sanicap),
+      sizeof (Device->MediaSanitize.SanitizeCapabilities)
+      );

```

Comment:
Good question. The original intent of the of the assert check is due to the fact that I'm comparing a 32 bit structure to a UINT32 (SanitizeCapabilities in the MEDIA_SANITIZE_PROTOCOL). Since the "sanicap" structure is an NVM Express construct, I was trying to leave out that level of detail in the MEDIA_SANITIZE_PROTOCOL header file. Therefore, I just made the field a UINT32. It's really just a sanity check to see if anyone decides to change the size of the "sanicap" structure for NVM Express, the correlating change in the agnostic MEDIA_SANITIZE_PROTOCOL must also be changed. If you feel strongly, I can add a run time check instead of an assert. Just let me know.

Reply:
Thanks for the explanation. I'm okay with leaving it as-is.